velox/exec/tests/RowContainerTest.cpp

```diff
@@ -304,23 +346,147 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
 
     std::vector<std::optional<T>> expectedOrder = {
         std::nullopt, lowest, 0.0, min, max, inf, nan};
+    setupFloatTestExpectedOutput<std::vector<std::optional<T>>>(
+        expectedOrder, ascending, nullsFirst);
+    VectorPtr expected = makeNullableFlatVector<T>(expectedOrder);
+
+    // Verify compare method with two rows as input
+    {
+      std::sort(rows.begin(), rows.end(), [&](const char* l, const char* r) {
+        return rowContainer->compare(l, r, 0, {nullsFirst, ascending}) < 0;
+      });
+      VectorPtr result = BaseVector::create(type, numRows, pool_.get());
+      rowContainer->extractColumn(rows.data(), numRows, 0, result);
+      assertEqualVectors(expected, result);
+    }
+
+    // Verify compare method with row and decoded vector as input
+    {
+      std::sort(
+          indexedRows.begin(),
+          indexedRows.end(),
+          [&](const std::pair<int, char*>& l, const std::pair<int, char*>& r) {
+            return rowContainer->compare(
+                       l.second,
+                       rowContainer->columnAt(0),
+                       decoded,
+                       r.first,
+                       {nullsFirst, ascending}) < 0;
+          });
+      auto result =
+          createResultVectorForSimpleComparison<T>(indexedRows, decoded);
+      assertEqualVectors(expected, result);
+    }
+
+    // Verify compareRows method with row as input.
```

Comment:
Is this different from the first test case (// Verify compare method with two rows as input) ? These seem identical.